controllers/recipe.js

```diff
@@ -0,0 +1,132 @@
+const { BadRequestError, NotFoundError } = require('../errors/index.js')
+const ObjectId = require('mongodb').ObjectId
+const Recipe = require('../models/Recipe.js')
+
+const getUserRecipes = async (req, res) => {
+    const { userId } = req.body
+    try {
+        const userRecipes = await Recipe.find({createdBy: userId})
+        res.json({userRecipes})
+    } catch (error) {
+        res.status(500).json({ message: 'Internal Server Error.' });
+    }
+}
+
+const getAllRecipes = async (req, res) => {
+    try {
+        const cuisineType = req.query.name
+        const allRecipes = await Recipe.find()
+        if (cuisineType) {
+            const filteredRecipes = allRecipes.filter(recipe => recipe.cuisineType + '=' === cuisineType);
+            res.json({filteredRecipes})
+        } else {
+            res.json({allRecipes})
+        }
+    } catch (error) {
+        res.status(500).json({ message: 'Internal Server Error.' });
+    }
+}
+
+const getRecipe = async (req, res) => {
+    try {
+        const recipe = await Recipe.findOne({
```

Comment:
You removed the line where you get the id from the request parameter, the id is undefined here.
Add this line back:
```
const id = req.params.id
```